tools/report_ci_failure.py

```diff
@@ -46,6 +49,8 @@ def report(self, branch, commit, infourl=None, job_name=None):
                 build logs.
             job_name (str): name of the failed ci job.
         """
+        if branch != 'master' or not self._stable_regex.search(branch):
```

Comment:
This should be `and` not `or`
```suggestion
        if branch != 'master' and not self._stable_regex.search(branch):
```